### PR TITLE
DI-344 Add new tests from NHSUK suite

### PR DESCRIPTION
**Pull request by Corbisc**
## Link to JIRA Ticket

[-](https://nhsd-jira.digital.nhs.uk/browse/DI-344)

## Description

This adds a series of integration tests that NHSUK have that we didn't have coverage for.

### Noteworthy Changes

- None of note

## Type of change

- Integration tests

## Testing

Please tick the testing that has been completed

- [ ] Unit tests
- [ ] Integration tests

## Developer Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the [code formatting checks](../README.md#code-quality)
- [ ] I have run the [code quality checks](../README.md#code-quality)
- [ ] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Integration tests)
- [ ] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Review by ed-sparkes — DISMISSED** [review]

File: test/integration/features/F001_Valid_Change_Events.feature [hunk 1]
```diff
@@ -41,3 +41,51 @@ Feature: F001. Ensure valid change events are converted and sent to DOS
     When the Changed Event is sent for processing with "valid" api key
     Then the processed Changed Request is sent to Dos
     And the Changed Request with changed address is captured by Dos
+
+@complete @kit
```
Comment (JackPlowman) [inline test/integration/features/F001_Valid_Change_Events.feature:45]: Can the kit tag be removed from this line

File: test/integration/steps/test_parent_steps.py [hunk 2]
```diff
@@ -552,3 +620,43 @@ def sequence_id_error_logs(context):
...
+@then("the opening times changes are marked as valid")
+def no_opening_times_errors(context):
+    query = (
+        f'fields message | sort @timestamp asc | filter correlation_id="{context["correlation_id"]}"'
+        ' | filter message like "Specified opening times not equal"'
+    )
+    logs = get_logs(query, "processor", context["start_time"])
+    assert logs != [], "ERROR!!.. log messages showing in cloudwatch."
+
+
```
Comment (JackPlowman) [inline test/integration/steps/test_parent_steps.py:644]: In `test/integration/steps/pytest.ini` can you add these lines under the pytest title
```
filterwarnings =
    ignore::DeprecationWarning
```

Comment (JackPlowman) [reply test/integration/steps/test_parent_steps.py:644]: This comment is unrelated to this file. But the deprecation warning is due to this `DeprecationWarning: invalid escape sequence \/`

**Review by JackPlowman — CHANGES_REQUESTED** [review]
Looks good, couple of minor changes to remove warnings

File: test/integration/features/F001_Valid_Change_Events.feature [hunk 3]
```diff
@@ -41,3 +41,51 @@ Feature: F001. Ensure valid change events are converted and sent to DOS
...
+  Scenario: F001S007. A valid change event with special characters is processed by DOS
```
Comment (JackPlowman) [inline test/integration/features/F001_Valid_Change_Events.feature:46]: Can all these tests be marked with `@dev` as well (if they can run on dev/task environments)

**Review by ed-sparkes — DISMISSED** [review]

**Review by mik3ola — DISMISSED** [review]

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Review by ed-sparkes — APPROVED** [review]

**Pull request merged**